### PR TITLE
fix bsp/display.h

### PR DESCRIPTION
**Pull request by tore-espressif**
* Added backlight functions to `bsp/display.h` so users can use them without including the main `bsp/esp-bsp.h`. This is important for future BSP version without LVGL
* Added capabilities of S3_eye board
* Add support for S3-EYE BOOT button

No need to release now

### CODE REVIEW

File: bsp/esp32_s2_kaluga_kit/include/bsp/display_backlight.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#pragma once
+#include "esp_err.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Backlight functions are not implemented - Kaluga board doesn't provide backlight control
+   These functions are here to provide consistent API with other Board Support Packages */
+esp_err_t bsp_display_backlight_on(void);
```
Comment (espzav) [inline bsp/esp32_s2_kaluga_kit/include/bsp/display_backlight.h:23]: Can we add here  `esp_err_t bsp_display_brightness_set(int brightness_percent);` only for compatibility?

**Review by espzav — COMMENTED** [review]
@tore-espressif Thank you for these changes. It is good idea to move these functions to another header, but why is it inside `display.h`? Do we really need new header file?

File: bsp/esp32_s2_kaluga_kit/include/bsp/display_backlight.h (hunk 1 above)
Comment (tore-espressif) [reply bsp/esp32_s2_kaluga_kit/include/bsp/display_backlight.h]: Done, thanks!

**Comment by tore-espressif** [comment]
@espzav Thank you for the review! At first I wanted the `bsp/display.h` to be 'semi-private'. So only users who don't want to use LVGL would include it. But now there are more public definitions so the `bsp/display_backlight.h` does not make sense anymore.

Removed!

**Review by espzav — APPROVED** [review]
LGTM

**Pull request merged**